solver/jobs.go

```diff
@@ -425,13 +425,15 @@ func (jl *Solver) loadUnlocked(v, parent Vertex, j *Job, cache map[Vertex]Vertex
 
 func (jl *Solver) connectProgressFromState(target, src *state) {
 	for j := range src.jobs {
-		if _, ok := target.allPw[j.pw]; !ok {
-			target.mpw.Add(j.pw)
-			target.allPw[j.pw] = struct{}{}
-			j.pw.Write(identity.NewID(), target.clientVertex)
-			if j.span != nil && j.span.SpanContext().IsValid() {
-				target.mspan.Add(j.span)
-			}
+		if _, ok := target.allPw[j.pw]; ok {
+			continue
+		}
+		target.mpw.Add(j.pw)
+		target.allPw[j.pw] = struct{}{}
+		j.pw.Write(identity.NewID(), target.clientVertex)
+		span := src.vtx.Span()
+		if span != nil && span.SpanContext().IsValid() {
+			target.mspan.Add(span)
```

Comment:
This `if span != nil` was getting very deeply nested in the original changes (when I was using an `atomic.Pointer`, hence the inversion of the `if`. The nesting isn't so bad now, so I could revert it back to the way it was if needed - but I prefer this version with the shallower nesting, personally.